src/pycram/worlds/multiverse_communication/clients.py

```diff
@@ -0,0 +1,568 @@
+        param constraint: The constraint.
+        return: The link names of the constraint.
+        """
+        return self.get_parent_link_name(constraint), self.get_constraint_child_link_name(constraint)
+
+    def get_parent_link_name(self, constraint: Constraint) -> str:
+        """
+        Get the parent link name of the constraint.
+        param constraint: The constraint.
+        return: The parent link name of the constraint.
+        """
+        return self.get_link_name_for_constraint(constraint.parent_link)
+
+    def get_constraint_child_link_name(self, constraint: Constraint) -> str:
+        """
+        Get the child link name of the constraint.
+        param constraint: The constraint.
+        return: The child link name of the constraint.
+        """
+        return self.get_link_name_for_constraint(constraint.child_link)
+
+    @staticmethod
+    def get_link_name_for_constraint(link: Link) -> str:
+        """
+        Get the link name from link object, if the link belongs to a one link object, return the object name.
+        param link: The link.
+        return: The link name.
+        """
+        return link.name if not link.is_only_link else link.object.name
+
+    def detach(self, constraint: Constraint) -> None:
+        """
+        Request to detach the child link from the parent link.
+        param constraint: The constraint.
+        """
+        parent_link_name, child_link_name = self.get_constraint_link_names(constraint)
+        self._detach(child_link_name, parent_link_name)
+
+    def _detach(self, child_link_name: str, parent_link_name: str) -> None:
+        """
+        Detach the child link from the parent link.
+        param child_link_name: The name of the child link.
+        param parent_link_name: The name of the parent link.
+        """
+        self._request_single_api_callback(API.DETACH, child_link_name, parent_link_name)
+
+    def _get_attachment_pose_as_string(self, constraint: Constraint) -> str:
+        """
+        Get the attachment pose as a string.
+        param constraint: The constraint.
+        return: The attachment pose as a string.
+        """
+        pose = constraint.parent_to_child_transform.to_pose()
+        return self._pose_to_string(pose)
+
+    @staticmethod
+    def _pose_to_string(pose: Pose) -> str:
+        """
+        Convert the pose to a string.
+        param pose: The pose.
+        return: The pose as a string.
+        """
+        return f"{pose.position.x} {pose.position.y} {pose.position.z} {pose.orientation.w} {pose.orientation.x} " \
+               f"{pose.orientation.y} {pose.orientation.z}"
+
+    def check_object_exists(self, obj: Object) -> bool:
+        """
+        Check if the object exists in the simulation.
+        param obj: The object.
+        return: Whether the object exists in the simulation.
+        """
+        return self._object_exists(obj.name)
+
+    def _object_exists(self, object_name: str) -> bool:
```

Comment:
You could also merge these two methods, since they are only one line each

Reply:
resolved in #177